homeassistant/components/sensor/github.py

```diff
@@ -0,0 +1,335 @@
+                    "number": _pr.number,
+                    "title": _pr.title,
+                    "url": _pr.html_url,
+                    "body": _pr.body,
+                    "labels": labels,
+                    "assignee": _pr.assignee,
+                    "milestone": _pr.milestone,
+                    "comments": _pr.comments,
+                    "user": {
+                        "login": _pr.user.login,
+                        "name": _pr.user.name,
+                        "avatar_url": _pr.user.avatar_url,
+                        "url": _pr.user.html_url
+                    }
+                })
+            self.open_pull_requests = json.dumps(open_pull_requests)
+
+            last_commit = repo.get_commits()[0]
+            self.last_commit = json.dumps({
+                "sha": last_commit.sha,
+                "message": last_commit.commit.message,
+                "url": last_commit.html_url,
+                "author": {
+                    "login": last_commit.author.login,
+                    "name": last_commit.author.name,
+                    "avatar_url": last_commit.author.avatar_url,
+                    "url": last_commit.author.html_url
+                }
+            })
+
+            releases = repo.get_releases()
+            if releases and releases.totalCount > 0:
+                self.latest_release = json.dumps({
+                    "title": releases[0].title,
+                    "tag": releases[0].tag_name,
+                    "url": releases[0].html_url,
+                    "created_at": str(releases[0].created_at),
+                    "published_at": str(releases[0].published_at),
+                    "body": releases[0].body
+                })
+
+            self.available = True
+        except self._github.BadCredentialsException as err:
+            _LOGGER.error("Bad Credentials for %s: %s",
+                          self.repository_path, err)
+            self.available = False
+        except self._github.UnknownObjectException as err:
+            _LOGGER.error("UnknownObjectException for %s: %s",
+                          self.repository_path, err)
+            self.available = False
+        except self._github.RateLimitExceededException as err:
+            _LOGGER.error("RateLimitExceededException for %s: %s",
+                          self.repository_path, err)
+            self.available = False
+        except self._github.BadAttributeException as err:
+            _LOGGER.error("BadAttributeException for %s: %s",
+                          self.repository_path, err)
+            self.available = False
+        except self._github.TwoFactorException as err:
+            _LOGGER.error("TwoFactorException for %s: %s",
+                          self.repository_path, err)
+            self.available = False
+        except self._github.GithubException as err:
+            _LOGGER.error("GithubException for %s: %s",
+                          self.repository_path, err)
```

Comment:
This is very verbose. Is there a `github` base exception that we could just catch?

If not, can you rearrange things so the long list of exceptions is at least not duplicated?